### ISSUE TITLE
[Feature Request] Multiple Task Learning

### ISSUE THREAD

**Issue by fangkuann** [tensorflow/ranking#85]
Hi, thanks very much for openning source this tensorflow ranking library! 
Since in the NN community, the ranking algorithm diverses a lot, for instance, one big branch is to learn from multiple targets simultaneously. 
Will tf-ranking library support multi-task learning? If will, what's the schedule.

**Comment by bendersky** [comment]
@fangkuann thank you for the feedback! We are currently doing some preliminary analysis of a multi-task learning setting. We do not have a concrete deadline yet, but we will keep this issue open to update on the progress.

**Comment by jeisinge** [comment]
Will the multitple task enhancement be compatible with [contrib.estimator.multi_head](https://www.tensorflow.org/api_docs/python/tf/contrib/estimator/multi_head) or [tensorflow_estimator head](https://github.com/tensorflow/estimator/blob/master/tensorflow_estimator/python/estimator/head/base_head.py)?

We are interested in adding in these kinds of out-of-the-box pointwise classification/regression heads to our ranking model.

**Comment by xuanhuiwang** [comment]
This show you an example about how to use multi-head in tf-ranking.

@jeisinge: No, the multi-task in tf-ranking is different from the head in tensorflow_estimator. But you can use the ranking_head for classification or regression training by using pointwise loss like sigmoid_cross_entropy in tf-ranking library.

**Comment by jeisinge** [comment]
@xuanhuiwang : cool - right on time.  Thank you!

I am still digesting this change and I might be reading it wrong... does this support 3+ heads?  Also, does this support a head for pairwise loss(`group_size = 2`) and a head for pointwise loss (`group_size = 1`)?

**Comment by xuanhuiwang** [comment]
No. group_size is needed only in the group_score_fn. 

Yes, it should support arbitrary number of heads. You can specific the loss_key to create pairwise loss such as pairwise_logistic_loss or pointwise loss such as sigmoid_cross_entropy_loss.
